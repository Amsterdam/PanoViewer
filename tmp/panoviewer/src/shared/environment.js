export const ENVIRONMENTS = {
  DEVELOPMENT: 'DEVELOPMENT',
  ACCEPTANCE: 'ACCEPTANCE',
  PRE_PRODUCTION: 'PRE_PRODUCTION',
  PRODUCTION: 'PRODUCTION'
};

export const HOSTS = {
  PRODUCTION: 'data.amsterdam.nl',
  PRE_PRODUCTION: 'pre.data.amsterdam.nl',
  ACCEPTANCE: 'acc.data.amsterdam.nl',
  DEVELOPMENT: 'localhost'
};

export const getEnvironment = (host) => {
  if (host === HOSTS.PRE_PRODUCTION) {
    return ENVIRONMENTS.PRE_PRODUCTION;
  }

  switch (process.env.NODE_ENV) {
    case 'production':
      return ENVIRONMENTS.PRODUCTION;
    case 'acceptance':
      return ENVIRONMENTS.ACCEPTANCE;
    default:
      return ENVIRONMENTS.DEVELOPMENT;
  }
};

// eslint-disable-next-line arrow-body-style
export const isDevelopment = () => {
  // istanbul ignore next // hard to test because window items are hard to mock
  return getEnvironment(window.location.hostname) === ENVIRONMENTS.DEVELOPMENT;
};

// This object enables/disables features that are not yet released
export const features = {
  eigendommen: true
};

const ENVIRONMENT = (process.env.NODE_ENV).toUpperCase();

export default ENVIRONMENT;
