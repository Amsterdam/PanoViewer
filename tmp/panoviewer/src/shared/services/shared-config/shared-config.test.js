import SHARED_CONFIG from './shared-config';
import { getEnvironment } from '../../environment';

jest.mock('../../environment');

describe('The sharedConfig service', () => {
  it('gives you the configuration based on global environment', () => {
    getEnvironment.mockImplementation(() => ('PRODUCTION'));
    expect(SHARED_CONFIG).toMatchSnapshot();

    getEnvironment.mockImplementation(() => ('PRE_PRODUCTION'));
    expect(SHARED_CONFIG).toMatchSnapshot();

    getEnvironment.mockImplementation(() => ('ACCEPTATION'));
    expect(SHARED_CONFIG).toMatchSnapshot();

    getEnvironment.mockImplementation(() => ('DEVELOPMENT'));
    expect(SHARED_CONFIG).toMatchSnapshot();
  });
});
