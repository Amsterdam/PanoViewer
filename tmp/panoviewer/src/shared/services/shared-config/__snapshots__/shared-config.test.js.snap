// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The sharedConfig service gives you the configuration based on global environment 1`] = `
Object {
  "API_ROOT": "https://acc.api.data.amsterdam.nl/",
  "AUTH_HEADER_PREFIX": "Bearer ",
  "CACHE_EXPIRATION": 0,
  "CACHE_THRESHOLD": 30,
  "PANORAMA_THUMBNAIL_URL": "panorama/thumbnail/",
  "RADIUS": 50,
  "ROOT": "https://acc.data.amsterdam.nl/",
  "ROOT_KEYS": Array [
    "API_ROOT",
  ],
  "THUMBNAIL_WIDTH": 240,
}
`;

exports[`The sharedConfig service gives you the configuration based on global environment 2`] = `
Object {
  "API_ROOT": "https://acc.api.data.amsterdam.nl/",
  "AUTH_HEADER_PREFIX": "Bearer ",
  "CACHE_EXPIRATION": 0,
  "CACHE_THRESHOLD": 30,
  "PANORAMA_THUMBNAIL_URL": "panorama/thumbnail/",
  "RADIUS": 50,
  "ROOT": "https://acc.data.amsterdam.nl/",
  "ROOT_KEYS": Array [
    "API_ROOT",
  ],
  "THUMBNAIL_WIDTH": 240,
}
`;

exports[`The sharedConfig service gives you the configuration based on global environment 3`] = `
Object {
  "API_ROOT": "https://acc.api.data.amsterdam.nl/",
  "AUTH_HEADER_PREFIX": "Bearer ",
  "CACHE_EXPIRATION": 0,
  "CACHE_THRESHOLD": 30,
  "PANORAMA_THUMBNAIL_URL": "panorama/thumbnail/",
  "RADIUS": 50,
  "ROOT": "https://acc.data.amsterdam.nl/",
  "ROOT_KEYS": Array [
    "API_ROOT",
  ],
  "THUMBNAIL_WIDTH": 240,
}
`;

exports[`The sharedConfig service gives you the configuration based on global environment 4`] = `
Object {
  "API_ROOT": "https://acc.api.data.amsterdam.nl/",
  "AUTH_HEADER_PREFIX": "Bearer ",
  "CACHE_EXPIRATION": 0,
  "CACHE_THRESHOLD": 30,
  "PANORAMA_THUMBNAIL_URL": "panorama/thumbnail/",
  "RADIUS": 50,
  "ROOT": "https://acc.data.amsterdam.nl/",
  "ROOT_KEYS": Array [
    "API_ROOT",
  ],
  "THUMBNAIL_WIDTH": 240,
}
`;
